extension/memorylimiterextension/memorylimiter_test.go

```diff
@@ -95,3 +96,8 @@ func TestMemoryPressureResponse(t *testing.T) {
 		})
 	}
 }
+
+func TestMiddleware(t *testing.T) {
```

Comment:
I think using `assert.Implements` would be better for this test. Here, if for some reason, the struct doesn't implement the interface anymore, we will get a panic before running the tests.